### PR TITLE
Satellite

### PR DESCRIPTION
**Pull request by 4e554c4c**

### CODE REVIEW

File: src/data.jl [hunk 1]
```diff
@@ -1,442 +1,33 @@
...
-const s23_v3 = 8.191520442889917986e-01
-export s23_v3
-const s23_per = 4.308204499999999825e04 # periodicity of Sat. 23 [s]
-export s23_per
-const s23_alt = 2.020000000000000000e07 # altitude of Sat. 23 [m]
-export s23_alt
-const s23_pha = 9.712388980384689674e00 # phase of Sat. 23 [rad]
-export s23_pha
+export DATA_FILE
+const DATA_FILE="data.dat"
```
Comment (coleschon) [inline src/data.jl:2]: I want this to read from STDIN so we could pass a different file in

File: src/satellite.jl [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
+                   $(lines[i+6]),
+                   $(lines[i+7]),
+                   $(lines[i+8])))
+        push!(sats, ex)
+    end
+    eval(:(const satellites = [$(sats...)]))
+end
+
+export Sat, satellites, position
+
```
Comment (coleschon) [inline src/satellite.jl:44]: A much cleaner way of doing this, have you tested?

File: src/data.jl (hunk 1 above)
Comment (coleschon) [reply src/data.jl:2]: And also have STDERR printout if the file is not found....
Do this and I'll approve.

**Review by coleschon — CHANGES_REQUESTED** [review]
Read DATA_FILE from STDIN with error checking and I'll approve.

File: src/satellite.jl (hunk 2 above)
Comment (4e554c4c) [reply src/satellite.jl:44]: yep! all tests pass
might be good to add in CI so this can be verified

File: src/data.jl (hunk 1 above)
Comment (4e554c4c) [reply src/data.jl:2]: that's not really how Peter's executables work though, they always read from `data.dat`

Comment (4e554c4c) [reply src/data.jl:2]: Also reading a `dat` from STDIN is just not possible since each of these programs take input on stdin

Comment (coleschon) [reply src/data.jl:2]: Ok that's fine then in that case

**Pull request merged**